src/parser/lexer/mod.rs

```diff
@@ -243,25 +243,39 @@ fn recognize_command_or_identifier(identifier: String) -> Token {
     }
 }
 
-fn is_valid_identifier_character(chr: char) -> bool {
-    is_valid_identifier_leading_character(chr) || chr.is_ascii_digit() || chr == '\\' || chr == '-'
-}
-
 fn is_valid_identifier_leading_character(chr: char) -> bool {
     chr.is_ascii_alphabetic() || chr == '_'
 }
 
+fn is_valid_identifier_end_character(chr: char) -> bool {
+    is_valid_identifier_leading_character(chr) || chr.is_ascii_digit()
+}
+
+fn is_valid_identifier_middle_character(chr: char) -> bool {
+    is_valid_identifier_end_character(chr) || chr == '-'
+}
+
 fn lex_identifier_raw(input: LexInput) -> LexResult<String> {
     expecting(
         "a valid identifier",
         map(
             tuple::<_, _, InternalLexError, _>((
                 take_while1(is_valid_identifier_leading_character),
-                take_while(is_valid_identifier_character),
+                take_while(is_valid_identifier_middle_character),
             )),
             |(left, right)| format!("{}{}", left, right),
         ),
     )(input)
+    .and_then(|(remaining, result)| {
+        if !result.ends_with(is_valid_identifier_end_character) {
+            Err(nom::Err::Failure(LexError::from_kind(
+                input,
+                LexErrorKind::ExpectedContext("valid identifier ending character"),
```

Comment:
Passing `input` here means the error will start at the beginning of the identifier.
```suggestion
                LexErrorKind::ExpectedContext("valid identifier"),
```